update partition-table.md (#3087)
pingcap/docs: partitioned-table.md

## Partitioning types

This section introduces the types of partitioning which are available in TiDB. Currently, TiDB supports Range partitioning and Hash partitioning.

Range partitioning is used to resolve the performance issues caused by a large amount of deletions in the application, and it supports fast drop partition operations. Hash partitioning is used to scatter the data when there are a large amount of writes.

### Range partitioning

> When a table is partitioned by Range, each partition contains rows for which the partitioning expression value lies within a given Range. Ranges have to be contiguous but not overlapping. You can define it by using the `VALUES LESS THAN`.

Reviewer: ```suggestion
When a table is partitioned by Range, each partition contains rows for which the partitioning expression value lies within a given Range. Ranges have to be contiguous but not overlapping. You can define it by using `VALUES LESS THAN`.
```

You can see that the inserted record `(NULL, 'mothra')` falls into the same partition as `(0, 'gigan')`.

> > **Attention**

Reviewer: ```suggestion
> **Note:**
```

> > The processing of NULL by Hash partition here is the same as [Document description of MySQL](https://dev.mysql.com/doc/refman/8.0/en/partitioning-handling-nulls.html), but it is not consistent with the actual behavior of MySQL. In other words, MySQL's documentation is not consistent with its implementation.

Reviewer: ```suggestion
> `NULL` values by Hash partitions in TiDB are handled in the same way as described in [How MySQL Partitioning Handles NULL](https://dev.mysql.com/doc/refman/8.0/en/partitioning-handling-nulls.html), which, however, is not consistent with the actual behavior of MySQL. In other words, MySQL's implementation in this case is not consistent with its documentation.
```

> 

> > The final behavior of TiDB is subject to the description in this document.

Reviewer: ```suggestion
> In this case, the actual behavior of TiDB is in line with the description of this document.
```

You can add a non-unique index by using `ALTER TABLE` statements. But if you want to add a unique index, the `c1` column must be included in the unique index.

> When using a partitioned table, the prefix index cannot be specified as a unique attribute:

Reviewer: ```suggestion
When using a partitioned table, you cannot specify the prefix index as a unique attribute:
```

+------|------+
5 rows in set (0.00 sec)
```

> The environment variable `tidb_enable_table_partition` can control whether to enable the partition table function. If this variable is set to `off`, the partition information will be ignored when the table is built, and the table will be built as a normal table.

Reviewer: ```suggestion
The `tidb_enable_table_partition` environment variable controls whether to enable the partitioned table feature. If this variable is set to `off`, the partition information will be ignored when a table is created, and this table will be created as a normal table.
```

> This variable is only used in table creation. After the table is created, it is invalid to modify this variable value. For details, see [TiDB specific system variables](/tidb-specific-system-variables.md#tidb_enable_table_partition).

Reviewer 2: ```suggestion
This variable is only used in table creation. After the table is created, modify this variable value takes no effect. For details, see [TiDB specific system variables](/tidb-specific-system-variables.md#tidb_enable_table_partition).
```

Reviewer 2: Rest LGTM

> This variable is only used to build the table, and it is invalid to modify the variable after the table has been built. For details, see [TiDB specific system variables] (/tidb-specific-system-variables.md#tidb_enable_table_partition).

Reviewer: ```suggestion
This variable is only used in table creation. After the table is created, it is invalid to modify this variable value. For details, see [TiDB specific system variables](/tidb-specific-system-variables.md#tidb_enable_table_partition).
```
